docs/documentation/helm.md

```diff
@@ -0,0 +1,276 @@
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: myModule
+spec:
+  replicas: {{ .Values.myModule.replicas }}
+```
+
+This is done transparently to users.
+
+##### Mapping user properties using JSONPath expressions
+
+Let's see the following YAML resource:
+
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: helm-on-kubernetes-example
+...
+```
+
+Dekorate will not replace the property at `metadata.name` with `{{ .Values.myModule.name }}`.
+However, Dekorate allows users to define [JSONPath](https://tools.ietf.org/id/draft-goessner-dispatch-jsonpath-00.html) expressions to map properties into the Helm values file. Let's see how to do it using the above example to map the property `metadata.name` with `{{ .Values.myModule.name }}`.
+
+**Note**: JSONPath works with JSON, not YAML. So, for testing your expressions, you first need to convert YAML files to JSON files using [this online tool](https://jsonformatter.org/yaml-to-json) and then use some JSONPath online evaluator like [https://jsonpath.com/](https://jsonpath.com/). Remember to wrap the json objects into lists.
+
+After getting familiar with JSONPath, the expression to map the metadata name value is `$..metadata.name`, so you need to add it to your configuration:
+
+```
+dekorate.helm.name=myChart
+dekorate.helm.description=Description of my Chart
+
+# Map all the metadata name resources
+dekorate.helm.values[0].property=myModule.name
+dekorate.helm.values[0].jsonPaths=$..metadata.name
+```
+
+The resulting `values.yaml` file will look like as:
+
+```yaml
+myModule:
+  name: helm-on-kubernetes-example
+```
+
+The `myModule.name` value is set automatically by Dekorate. However, users can provide other values using the `value` property:
+
+```
+dekorate.helm.name=myChart
+dekorate.helm.description=Description of my Chart
+
+# Map all the metadata name resources
+dekorate.helm.values[0].property=myModule.name
+dekorate.helm.values[0].jsonPaths=$..metadata.name
+## Overwrite value:
+dekorate.helm.values[0].value=this-is-another-name
+```
+
+And the `values.yaml` file will now contain:
+
+```yaml
+myModule:
+  name: this-is-another-name
+```
+
+##### Mapping multiple properties at once
+
+What about if the properties are located in different places, for example:
+
+```yaml
+---
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: helm-on-kubernetes-example ## we need to map this property
+spec:
+  rules:
+    - host: my-host
+      http:
+        paths:
+          - backend:
+              service:
+                name: helm-on-kubernetes-example ## And this property
+                port:
+                  name: http
+            path: /
+            pathType: Prefix
+```
+
+For this scenario, you need to provide a comma-separated list of JSONPath expressions to be mapped to the same property `myModule.name`:
```

Comment:
Same remark as before. Transition text is still missing as the user still don't know what you would like to do here.

Reply:
I see what you meant now. Updated.